server/routes/posts.js

```diff
@@ -4,9 +4,9 @@ import auth from '../middleware/auth.js';
 
 const router = express.Router();
 
+router.get('/search', getPostsBySearch);
```

Comment:
The order of these routers are very crucial. If you change the order, it maybe causes some confections about APIS.